packages/odyssey-react-mui/src/theme/useContrastMode.test.tsx

```diff
@@ -0,0 +1,254 @@
+  });
+
+  describe("useContrastMode hook", () => {
+    it("should return lowContrast mode by default", () => {
+      const { result } = renderHook(() => useContrastMode({}), {
+        wrapper: createWrapper({ contrastMode: "lowContrast" }),
+      });
+
+      expect(result.current.contrastMode).toBe("lowContrast");
+      expect(result.current.contrastContainerRef.current).toBe(null);
+      expect(result.current.parentBackgroundColor).toBe("#ffffff");
+    });
+
+    it("should respect explicitly set contrast mode", () => {
+      const { result } = renderHook(
+        () => useContrastMode({ contrastMode: "highContrast" }),
+        {
+          wrapper: createWrapper({ contrastMode: "lowContrast" }),
+        },
+      );
+
+      expect(result.current.contrastMode).toBe("highContrast");
+    });
+
+    it("should update contrast mode based on background color changes", async () => {
+      const TestComponent = () => {
+        const { contrastContainerRef, contrastMode } = useContrastMode({});
+        return (
+          <div ref={contrastContainerRef} data-testid="container">
+            {contrastMode}
+          </div>
+        );
+      };
+
+      // Mock getComputedStyle before rendering
+      window.getComputedStyle = jest.fn().mockImplementation(() => ({
+        backgroundColor: Tokens.HueNeutral50,
+      }));
+
+      const { getByTestId } = render(
+        <ContrastModeContext.Provider value={{ contrastMode: "lowContrast" }}>
+          <TestComponent />
+        </ContrastModeContext.Provider>,
+      );
+
+      const testContainer = getByTestId("container");
+
+      act(() => {
+        testContainer.style.backgroundColor = Tokens.HueNeutral50;
+        const event = new Event("transitionend");
+        Object.defineProperty(event, "propertyName", {
+          value: "background-color",
+        });
+        testContainer.dispatchEvent(event);
+      });
+
+      // Add a small delay to allow for state updates
+      await act(async () => {
+        await new Promise((resolve) => setTimeout(resolve, 0));
+      });
+
+      await waitFor(
+        () => {
+          expect(getByTestId("container").textContent).toBe("highContrast");
+        },
+        { timeout: 1000 },
+      );
+    });
+  });
+
+  describe("getBackgroundColor", () => {
+    beforeEach(() => {
+      window.getComputedStyle = jest
+        .fn()
+        .mockImplementation((element: HTMLElement) => ({
+          backgroundColor: element.style.backgroundColor || "rgba(0, 0, 0, 0)",
+        }));
+    });
+
+    afterEach(() => {
+      window.getComputedStyle = originalGetComputedStyle;
+    });
+
+    it("returns the background color of the element if it is not transparent", () => {
+      const element = document.createElement("div");
+      element.style.backgroundColor = "rgb(255, 0, 0)";
+      const result = getBackgroundColor(element);
+      expect(result).toBe("rgb(255, 0, 0)");
+    });
+
```

Comment:
I like the first test to be the default case, so in this case, if it's transparent.